### ISSUE TITLE
[lldb-dap] Enabling LLDB logging to a file does not work in lldb-dap

### ISSUE THREAD

**Issue by ashgti** [llvm/llvm-project#78334]
If you are using lldb/tools/lldb-dap and you try to enable logging to a file using something like:

```
lldb> log enable -f /tmp/my_log.txt llvm api
```

The file is never created and logging is not configured. Instead the logs are sent over the DAP as `Console` messages which may overwhelm the DAP with a large amount of logging, see https://github.com/llvm/llvm-project/blob/32dd5b20973bde1ef77fa3b84b9f85788a1a303a/lldb/tools/lldb-dap/lldb-dap.cpp#L1520

**Comment by ashgti** [comment]
@walter-erquinigo Do you know if the logging callback is required for Windows support? The original commit https://github.com/llvm/llvm-project/commit/9bcaf6ddfe34943e6ae6a319097524c117908913 was around Windows output redirect support but I'm not sure if the logging callback is required for windows or not.

**Comment by walter-erquinigo** [comment]
For context, lldb-dap has to do stdout/stderr redirection because there are some cases in which LLDB outputs error messages directly to the stdout/stderr, which cause the debug session to terminate unless we do the redirection.

The fact that `lldb> log enable -f /tmp/my_log.txt llvm api` is being redirected seems to be an issue with file descriptors in the `Debugger::EnableLog` function or somewhere deeper. 

Sadly, we can't just disable stdout/stderr redirection :(

**Comment by ashgti** [comment]
I don't think we need to disable the stdout/stderr redirection, I meant to not use the SBDebugger::Create() with logger callback. When a logger callback is set then it will always call the callback instead of using a file.

**Comment by ashgti** [comment]
This is actually fixed at head due to some recent refectoring in lldb-dap.
